test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java

```diff
@@ -251,6 +253,22 @@ public void execute(Runnable command) {
         };
     }
 
+    /**
+     * Whether the direct executor should be used for the executor with the given name. If yes, tasks submitted to that executor
+     * will be executed immediately inline instead of being scheduled in the deterministic task queue.
+     */
+    protected boolean shouldUseDirectExecutor(String executorName) {
+        return false;
+    }
+
+    /**
+     * Determine whether the task should be schedule right away instead of after the given delay.
+     * @return true if the task was scheduled right away, false otherwise.
+     */
+    protected boolean shouldScheduleNow(Runnable command, TimeValue delay, Executor executor) {
+        return false;
+    }
```

Comment:
This is needed to support task to run as part of `runAllRunnableTasks` without the need to avdance time. It is needed for translog upload on the stateless side.